File: src/java/src/main/java/dab/Point.java
package dab;

public record Point(int x, int y) {
    public boolean inBounds(int nx, int ny) {
        return x >= 0 && x <= nx && y >= 0 && y <= ny;
    }

    public int manhattan(Point other) {
        return Math.abs(x - other.x) + Math.abs(y - other.y);
    }
}

File: src/csharp/DotsAndBoxes.Core/Point.cs
namespace DotsAndBoxes;

public readonly record struct Point(int X, int Y) : IComparable<Point>
{
    public bool InBounds(int nx, int ny) => X >= 0 && X <= nx && Y >= 0 && Y <= ny;

    public int Manhattan(Point other) => Math.Abs(X - other.X) + Math.Abs(Y - other.Y);

    public int CompareTo(Point other)
    {
        var c = X.CompareTo(other.X);
        return c != 0 ? c : Y.CompareTo(other.Y);
    }
}
